controller/share_manager_controller.go

```diff
@@ -407,22 +407,32 @@ func (c *ShareManagerController) isShareManagerRequiredForVolume(sm *longhorn.Sh
 
 func (c *ShareManagerController) createShareManagerAttachmentTicket(sm *longhorn.ShareManager, va *longhorn.VolumeAttachment) {
 	log := getLoggerForShareManager(c.logger, sm)
+	nodeID := ""
+	pod, err := c.ds.GetPod(types.GetShareManagerPodNameFromShareManagerName(sm.Name))
```

Comment:
Use `GetPodRO` instead.

Reply:
Done.  I also switched the call in isResponsibleFor() because it also uses the pod as just a read-only object.